### DISCUSSION TITLE
BiCGStab memory footprint estimation

### DISCUSSION THREAD

**Discussion by tpadioleau** [ginkgo-project/ginkgo#1644] [General]
Hi, I am using the BiCGStab algorithm, I was wondering if you could give an estimation of the maximum memory footprint needed by Ginkgo during the `apply` call ? In function of the parameters of the problem:
- the size of the matrix,
- the number of right-hand sides,
- no preconditioning,
- a convergence logger.

Thanks a lot,
Thomas

**Comment by pratikvn** [comment]
Does this comment block maybe help: https://github.com/ginkgo-project/ginkgo/blob/ded657c87a7f01f228dbf7a730a1180405b2f567/core/solver/bicgstab.cpp#L144  ?

**Reply by tpadioleau** [reply]
I think so but can you clarify the meaning of "values" ? Is it the number of rhs ?

**Reply by pratikvn** [reply]
Yes, I think so. `values = nrhs * fp_bytes`, where `nrhs` is the number of rhs and `fp_bytes` is the number of bytes used for the floating point calculations (except when accessor is used), 4 for float, and 8 for double.

The storage for the matrix will depend on the matrix format used.

**Reply by tpadioleau** [reply]
Thanks ok I understand. Is this `* values` missing for the rest of the operations ? The second line says

> 31n * values + 2 * matrix/preconditioner storage

I guess it is supposed to be the summation of all the operations mentioned below ?

**Reply by pratikvn** [reply]
Yes. It seems to be missing. `31n * values + ...` is the summation of all the operations, so that is the one you need (and that should be correct). When not using a preconditioner, we just do a copy between vectors. So, you can replace the preconditioning operation (2n * values + storage) with a copy instead, which should be `2n * values`.

**Comment by tpadioleau** [comment]
Thanks a lot
